### PR TITLE
Pulls/2.0/convert to ss4

### LINKED ISSUE #1: Update codebase for SS4
Use namespace `SilverStripe\DocumentConverter`.

Sub-tasks can be added separately or as items in a checklist on this issue.

### PR DESCRIPTION
**Pull request by NightJar**
resolves #1

### CODE REVIEW

File: src/ImportField.php [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+
+        restore_error_handler();
+        if ($documentImporterFieldError) {
+            return ['error' => $documentImporterFieldError];
+        }
+
+        foreach ($this->unusedChildren as $child) {
+            $origStage = Versioned::current_stage();
+
+            Versioned::reading_stage('Stage');
```
Comment (robbieaverill) [inline src/ImportField.php]:
```diff
- Versioned::reading_stage('Stage')
+ Versioned::set_stage(Versioned::DRAFT);
```

File: src/ImportField.php [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+            unset($this->unusedChildren[$page->ID]);
+            file_put_contents(ASSETS_PATH . '/index-' . $sort . '.html', $this->getBodyText($subdoc, $subnode));
+
+            if ($sort) {
+                $page->Sort = $sort;
+            }
+            $page->Content = $this->getBodyText($subdoc, $subnode);
+            $page->write();
+            if ($publishPages) {
+                $page->doPublish();
```
Comment (robbieaverill) [inline src/ImportField.php]: Replace with `publishRecursive()`

File: src/ServiceConnector.php [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        $username = $this->config()->get('password');
+        if ($username) {
+            return $username;
+        }
+        $username = Environment::getEnv('DOCVERT_PASSWORD');
+        if ($username) {
+            return $username;
+        }
+        return null;
+    }
```
Comment (robbieaverill) [inline src/ServiceConnector.php]: We should consolidate this logic so we don't repeat it

File: src/ServiceConnector.php [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+    public function __construct($fileDescriptor, $chosenFolderID = null)
+    {
+        $this->fileDescriptor = $fileDescriptor;
+        $this->chosenFolderID = $chosenFolderID;
+    }
+
+    public function setUsername($username = null)
+    {
+        $this->docvertDetails['username'] = $username;
```
Comment (robbieaverill) [inline src/ServiceConnector.php:99]: Chaining would be nice 👍

File: src/ServiceConnector.php [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+            return $username;
+        }
+        return null;
+    }
+
+    public function import()
+    {
+        $ch = curl_init();
+
+        // PHP 5.5+ introduced CURLFile which makes the '@/path/to/file' syntax deprecated.
```
Comment (robbieaverill) [inline src/ServiceConnector.php]: PHP 5.6 is the minimum here, so don't need this check

File: tests/Stubs/TestPage.php [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+use SilverStripe\Dev\TestOnly;
+use SilverStripe\DocumentConverter\PageExtension;
+use Page;
+
+class TestPage extends Page implements TestOnly
+{
+    private static $extensions = [PageExtension::class];
+    private static $defaults = [
+        'Content' => '<h1>Default TestPage</h1><p>With pre-import content.</p>'
+    ];
```
Comment (robbieaverill) [inline tests/Stubs/TestPage.php]: Add a `$table_name`, otherwise 4.0.0 will throw warnings for not having it (silly, I know)

**Review by robbieaverill — CHANGES_REQUESTED** [review]
Couple of minor points

File: src/ServiceConnector.php (hunk 4 above)
Comment (NightJar) [reply src/ServiceConnector.php:99]: ⛓ 'd :)

**Review by robbieaverill — APPROVED** [review]

**Pull request merged**